pyts2/pipeline/base.py

```diff
@@ -27,15 +27,15 @@ def process_image(self, image):
             image = step.process_image(image)
         return image
 
-    def process_from(self, input, ncpus=1):
+    def process_from(self, input_data, ncpus=1):
         if ncpus > 1:
-            # FIXME: support parallel processing by setting up a pool and
-            # replacing map with pool.imap
-            map = map
-        yield from map(self.process_one, input)
+            from multiprocessing import Pool
+            pool = Pool(ncpus)
```

Comment:
I think pool would need to be stored in self, otherwise it could get GC'd?

Reply:
no because its still within the scope of that member it wouldnt be GCed until the iterator completes, I think.

Reply:
Sweet!